### DISCUSSION TITLE
Slow query result retrieval

### DISCUSSION THREAD

**Discussion by lrweck** [dbeaver/dbeaver#13844] [Q&A]
#### System information: 
- Windows 10 - v19041.572
- DBeaver 7.2.2.202010041557

#### Connection specification:
- PostgreSQL 12.4
- DBeaver default driver
- Using tunnels over SSH.

#### Describe the problem you're observing:

Running count(*) on a large table takes about 20s, and using "Explain Analyze" returns in about 6 seconds.

The query is:

`   select count(*) filter (where current_date - retdataca <= 30),
          count(*) filter ( where retdataca >= current_date - 30)
     from retorno `

Where table "retorno" has 64249009 records.

Since it only returns 1 tuple, I was expecting the execution time to be about the same as running Explain Analyze or running locally.

#### Steps to reproduce, if exist:

Create large table, with many records. Access database remotely, Run count(*) on large table.

**Comment by lrweck** [comment]
Proof:

Running the query:
![image](https://user-images.githubusercontent.com/26086198/96137253-d25a8200-0ed2-11eb-99a5-b574a190ef91.png)

Running explain analyze:
![image](https://user-images.githubusercontent.com/26086198/96137324-e9996f80-0ed2-11eb-9158-a4cf60c512de.png)

Running explain analyze locally, on PSQL:
![image](https://user-images.githubusercontent.com/26086198/96137461-0d5cb580-0ed3-11eb-9da7-1ac8c6fb133e.png)

Running the query locally on PSQL:
![image](https://user-images.githubusercontent.com/26086198/96137567-2c5b4780-0ed3-11eb-84d0-6af68da7d649.png)


I would think connection speed is negligible here, because there is very little data being transfered.

Besides, running SELECT on a single tuple returns really fast.

**Comment by lrweck** [comment]
@serge-rider were you able to reproduce it? Is there something else I can help with or maybe some other example?

**Comment by serge-rider** [comment]
Yes. But this still looks like PostgreSQL-specific question but not DBeaver-specific.
There are quite a lot of similar questions over the internet. For instance:
See https://stackoverflow.com/questions/3420563/postgres-explain-analyze-is-much-faster-than-running-the-query-normally

**Comment by lrweck** [comment]
> Yes. But this still looks like PostgreSQL-specific question but not DBeaver-specific.
> There are quite a lot of similar questions over the internet. For instance:
> See https://stackoverflow.com/questions/3420563/postgres-explain-analyze-is-much-faster-than-running-the-query-normally

The link you posted talks about OS page cache and network delay. 

Running the query locally, gives approximate results, however remotely, the difference is enormous. That must mean network, right? However the network part refers to data already processed by postgres and just being sent back. I can't understand how 1 row of result takes 15 seconds. Both connections transfer at above 5 MB/s (up and down) and latency is about 100 ms
